src/SalesShareDistributor.sol

```diff
@@ -0,0 +1,180 @@
+     * @param tokenContract FractionalizedToken     the fraction token
+     * @param listingId     uint256     the listing id on Schmackoswap
+     * @param permissioner  IPermissioner   the permissioner that permits claims
+     */
+    function afterSale(FractionalizedToken tokenContract, uint256 listingId, IPermissioner permissioner) external {
+        if (_msgSender() != tokenContract.issuer()) {
+            revert OnlyIssuer();
+        }
+
+        Metadata memory metadata = tokenContract.metadata();
+        (, uint256 ipnftId,,, IERC20 _paymentToken, uint256 askPrice, address beneficiary, ListingState listingState) =
+            schmackoSwap.listings(listingId);
+
+        if (listingState != ListingState.FULFILLED) {
+            revert ListingNotFulfilled();
+        }
+        if (ipnftId != metadata.ipnftId) {
+            revert ListingMismatch();
+        }
+        if (beneficiary != address(this)) {
+            revert InsufficientBalance();
+        }
+
+        _startClaimingPhase(tokenContract, listingId, _paymentToken, askPrice, permissioner);
+    }
+
+    //todo: ensure that no one can withdraw arbitrary amounts here
+    //by simply creating a new fractionalization and claim an arbitrary value
+
+    /**
+     * @notice When the sales beneficiary has not been set to the underlying erc20 token address but to the original owner's wallet instead,
+     *         they can invoke this method to start the claiming phase manually. This e.g. allows sales off the record.
+     *
+     *         Requires the originalOwner to behave honestly / in favor of the fraction holders
+     *         Requires the caller to have approved `price` of `paymentToken` to this contract
+     *
+     * @param   tokenContract FractionalizedToken  the fraction token contract
+     * @param   paymentToken  IERC20   the payment token contract address
+     * @param   paidPrice         uint256  the price the NFT has been sold for
+     * @param permissioner  IPermissioner   the permissioner that permits claims
+     */
+    function afterSale(FractionalizedToken tokenContract, IERC20 paymentToken, uint256 paidPrice, IPermissioner permissioner) external nonReentrant {
+        if (_msgSender() != tokenContract.issuer()) {
+            revert OnlyIssuer();
+        }
+
+        Metadata memory metadata = tokenContract.metadata();
+
+        //create a fake (but valid) schmackoswap listing id
+        uint256 fulfilledListingId = uint256(
+            keccak256(
+                abi.encode(
+                    SchmackoSwap.Listing(
+                        IERC1155Supply(address(0)), //this should be the IPNFT address
+                        metadata.ipnftId,
+                        _msgSender(),
+                        uint256(1),
+                        paymentToken,
+                        paidPrice,
+                        address(this),
+                        ListingState.FULFILLED
+                    ),
+                    block.number
```

Comment:
I don't understand why we still carry the schmackoSwap contract around with us. At the moment, we require the fraction holders to have trust in the original owner in both cases (1. owner sells ipnft via schmackosawp, 2. owner sells somewhere else, creates a fake schmackoswap listing via this contract and transfers the tokens to the distributor). 
I don't feel like we gain anything by having this schmackoswap dependency and could just only build out the 2nd case w/o any schmackoswap interation.

Reply:
Just leave it in because it's convenient and its sales *can* be trusted. You perspectively need to activate the claiming phase automatically after the original asset has been sold. Schmackoswap isn't controlled by any external entity so this is just a good first approach. Nobody needs to call this method :)